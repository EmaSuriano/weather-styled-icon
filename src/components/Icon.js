import styled from 'styled-components';
import PropTypes from 'prop-types';

const Icon = styled.div`
  position: relative;
  display: inline-block;
  height: 10em;
  width: 10em;
  font-size: ${props => `${props.size}em`};
`;

Icon.propTypes = {
  size: PropTypes.number,
};

Icon.defaultProps = {
  size: 1,
};

export default Icon;
