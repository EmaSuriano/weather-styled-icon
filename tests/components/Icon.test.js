import React from 'react';
import Icon from '../../src/components/Icon';
import { shallow } from 'enzyme';

describe('<Icon />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Icon />);
  });

  it('should render WeatherContainer', () => {
    const wrapper = shallow(<Icon size={1} />);

    expect(wrapper).toMatchSnapshot();
  });
});
