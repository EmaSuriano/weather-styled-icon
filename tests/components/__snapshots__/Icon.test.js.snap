// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should render WeatherContainer 1`] = `
<div
  className="sc-bdVaJa iXOemQ"
  size={1}
/>
`;
